Modules/Package.swift

```diff
@@ -11,10 +11,48 @@ let package = Package(
         .iOS(.v15),
     ],
     products: [
+        // The dependencies used only by the app (linked statically)
+        .library(name: "WordPressAppDependencies", targets: ["WordPressAppDependencies"]),
+        // The dependencies shared across all targets (linked dynamically to avoid duplication)
+        .library(name: "WordPressSharedDependencies", type: .dynamic, targets: ["WordPressSharedDependencies"]),
         .library(name: jetpackStatsWidgetsCoreName, targets: [jetpackStatsWidgetsCoreName]),
         .library(name: designSystemName, targets: [designSystemName]),
     ],
+    dependencies: [
+        .package(url: "https://github.com/Automattic/Gravatar-SDK-iOS", from: "2.1.0"),
+        .package(url: "https://github.com/wordpress-mobile/WordPress-iOS-Shared", branch: "task/remove-swiftlint"),
+        .package(url: "https://github.com/wordpress-mobile/WordPressUI-iOS", branch: "kean-patch-1"),
+        .package(url: "https://github.com/Automattic/Gridicons-iOS", branch: "develop"),
+        .package(url: "https://github.com/CocoaLumberjack/CocoaLumberjack", from: "3.8.5"),
+        .package(url: "https://github.com/johnxnguyen/Down", branch: "master"),
+        .package(url: "https://github.com/weichsel/ZIPFoundation", from: "0.9.19"),
+        .package(url: "https://github.com/tonymillion/Reachability", from: "3.7.5"),
+        .package(url: "https://github.com/wordpress-mobile/MediaEditor-iOS", branch: "task/spm-support"),
+        .package(url: "https://github.com/wordpress-mobile/FSInteractiveMap", from: "0.3.0"),
+        .package(url: "https://github.com/kaishin/Gifu", from: "3.4.1"),
+        .package(url: "https://github.com/SVProgressHUD/SVProgressHUD", from: "2.3.1"),
+        .package(url: "https://github.com/wordpress-mobile/NSObject-SafeExpectations", from: "0.0.6"),
+    ],
     targets: [
+        .target(name: "WordPressAppDependencies", dependencies: [
+            .product(name: "Gravatar", package: "Gravatar-SDK-iOS"),
+            .product(name: "GravatarUI", package: "Gravatar-SDK-iOS"),
+            .product(name: "MediaEditor", package: "MediaEditor-iOS"),
+            .product(name: "FSInteractiveMap", package: "FSInteractiveMap"),
+            .product(name: "Gifu", package: "Gifu"),
+        ]),
+        .target(name: "WordPressSharedDependencies", dependencies: [
```

Comment:
We probably need a bit more granular with these because, for example, Gridicons has a large bundle (1.8 MB) and is not needed in every target. Xcode does copy these bundles in every target even for dynamically linked frameworks.